### PR TITLE
Refactor environment 

### PR DESCRIPTION
**Pull request by ncsilvaa**
At this stage, we refactor all the data environment to clean old codes and improve usability and readability.
Moreover, we also add some behave tests to validate these new changes.

### CODE REVIEW

File: app/settings/agents_variables.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ICTRTS:
+  validation_size: 0.1
```
Comment (heitor57) [inline app/settings/agents_variables.yaml:2]: is this right?

File: app/settings/dataset_loaders.yaml [hunk 2]
```diff
@@ -308,7 +325,7 @@
       test_consumes: 5  
     splitting:
       strategy: temporal
-      train_size: 0.8
+      tInforain_size: 0.8
```
Comment (heitor57) [inline app/settings/dataset_loaders.yaml:328]: fix

File: app/settings/agents_variables.yaml (hunk 1 above)
Comment (thiagodks) [reply app/settings/agents_variables.yaml:2]: No, we forgot to remove it, it's just an idea

File: irec/environment/loader/train_test.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
+from typing import TypedDict, Tuple
+import numpy as np
+from irec.environment.dataset import Dataset
+
+TrainDatasetType = TypedDict('TrainDatasetType', {'path': str, 'file_delimiter': str, 'skip_head': bool})
+TestDatasetType = TypedDict('TestDatasetType', {'path': str, 'file_delimiter': str, 'skip_head': bool})
+ValidationDatasetType = TypedDict('ValidationDatasetType', {'path': str, 'file_delimiter': str, 'skip_head': bool})
+DatasetType = TypedDict('DatasetType', {'train': TrainDatasetType, 'test': TestDatasetType})
```
Comment (CarlosMito) [inline irec/environment/loader/train_test.py:8]: Will the syntax change to class-like?

**Review by CarlosMito — APPROVED** [review]

File: irec/environment/loader/train_test.py (hunk 3 above)
Comment (thiagodks) [reply irec/environment/loader/train_test.py:8]: Nops

**Review by heitor57 — APPROVED** [review]

**Pull request merged**